proxygen/lib/http/codec/compress/experimental/simulator/CompressionSimulator.cpp

```diff
@@ -172,23 +206,19 @@ void CompressionSimulator::setupRequest(uint16_t index, HTTPMessage&& msg,
   };
   callbacks_.emplace_back(index, decodeCompleteCB);
 
-  // Schedule the encode event
-  scheduleEvent([index, this, scheme] {
-      auto schemeIndex = scheme->index;
-      auto encodeRes = encode(scheme, index);
-      bool allowOOO = encodeRes.first;
-      if (schemeIndex < minOOOThresh()) {
-        allowOOO = false;
-        auto ack = scheme->getAck(schemeIndex);
-        if (ack) {
-          scheme->recvAck(std::move(ack));
-        }
-      }
-      stats_.allowedOOO += (allowOOO) ? 1 : 0;
-      scheme->encodedBlocks.emplace_back(
-        allowOOO, std::move(encodeRes.second), &callbacks_[index]);
-      eventBase_.runInLoop(scheme, true);
-    }, encodeDelay);
+  // Assume that all packets with same encodeDelay will form a packet
+  // train.  Encode them as a group, so can we can emulate packet
+  // boundaries more realistically, telling the encoder which blocks
+  // start such trains.
+  if (scheme->packetIndices.size() > 0) {
+    auto delayFromPrevious = encodeDelay - scheme->prev;
+    VLOG(1) << "request " << index << " delay " << delayFromPrevious.count();
+    if (delayFromPrevious > std::chrono::milliseconds(5)) {
```

Comment:
So "the same" encode delay really means within 5ms.  I assume this is to account for some jitter in the HAR file?

Reply:
Yes.  I tried both 1ms and 5ms.   5ms did result  in fewer and larger packets, but it didn't make a big difference on result.    I kind of forgot about it.   I think I was intending to switch it back to 1ms...

Reply:
I'll put this to 1ms in the merge